### PR TITLE
feat: render ignore settings [IDE-173]

### PR DESCRIPTION
**Pull request by cat2608**
### Description

IDE-173: This PR renders conditionally the Ignore settings in the UI

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 1]
```diff
@@ -98,8 +98,13 @@ class SnykProjectSettingsConfigurable(val project: Project) : SearchableConfigur
             snykProjectSettingsService?.additionalParameters = snykSettingsDialog.getAdditionalParameters()
         }
 
-        val params = DidChangeConfigurationParams(LanguageServerWrapper.getInstance().getSettings())
-        LanguageServerWrapper.getInstance().languageServer.workspaceService.didChangeConfiguration(params)
+        val wrapper = LanguageServerWrapper.getInstance()
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:101]: With my pending PR, we'll be able to completely move this to LanguageServerWrapper, as it has now a settings listener.

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 2]
```diff
@@ -172,7 +172,20 @@ class SnykLanguageClient : LanguageClient {
     private fun getSnykCodeResult(project: Project, snykScan: SnykScanParams): Map<SnykCodeFile, List<ScanIssue>> {
         check(snykScan.product == "code") { "Expected Snyk Code scan result" }
         if (snykScan.issues.isNullOrEmpty()) return emptyMap()
-        val map = snykScan.issues
+
+        val includeIgnoredIssues = pluginSettings().ignoredIssuesEnabled
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:176]: pluginSettings() should be extracted to a variable

**Review by bastiandoetsch — APPROVED** [review]
Looks good!

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt (hunk 2 above)
Comment (cat2608) [reply src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt]: Done 🙌

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 3]
```diff
@@ -98,8 +99,13 @@ class SnykProjectSettingsConfigurable(val project: Project) : SearchableConfigur
...
+        val params = DidChangeConfigurationParams(wrapper.getSettings())
+        wrapper.languageServer.workspaceService.didChangeConfiguration(params)
+
+        runBackgroundableTask("Updating Snyk Code settings", project, true) {
+            settingsStateService.isGlobalIgnoresFeatureEnabled = wrapper.getFeatureFlagStatus("snykCodeConsistentIgnores")
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:107]: Here the guard is missing that you only call LS when `isSnykCodeLSEnabled() == true`

File: src/main/kotlin/snyk/common/lsp/LanguageServerWrapper.kt [hunk 4]
```diff
@@ -119,6 +120,9 @@ class LanguageServerWrapper(
         } finally {
             isInitializing = false
         }
+
+        // update feature flags
+        pluginSettings().isGlobalIgnoresFeatureEnabled = getFeatureFlagStatus("snykCodeConsistentIgnores")
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/LanguageServerWrapper.kt:125]: here we need the guard as well

File: src/main/kotlin/snyk/common/lsp/LanguageServerWrapper.kt [hunk 5]
```diff
@@ -225,6 +229,33 @@ class LanguageServerWrapper(
         }
     }
 
+    fun getFeatureFlagStatus(featureFlag: String): Boolean {
+        ensureLanguageServerInitialized()
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/LanguageServerWrapper.kt:233]: Or you could put the guard here

File: src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt [hunk 6]
```diff
@@ -172,7 +173,21 @@ class SnykLanguageClient : LanguageClient {
...
         check(snykScan.product == "code") { "Expected Snyk Code scan result" }
         if (snykScan.issues.isNullOrEmpty()) return emptyMap()
-        val map = snykScan.issues
+
+        val pluginSettings = pluginSettings()
+        val includeIgnoredIssues = pluginSettings.ignoredIssuesEnabled
+        val includeOpenedIssues = pluginSettings.openIssuesEnabled
+
+        // get enablement status from settings
+        val isFeatureFlagEnabled = isFeatureFlagEnabled()
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt:182]: this should just refer to the ignore setting

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt (hunk 3 above)
Comment (cat2608) [reply src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt]: Done, please double check 🙏

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,6 +10,10 @@
 - (LS Preview) Fix long-running UI operation to run outside of UI thread
 - Remove duplicated annotations in Snyk Code
 
+## [2.8.1]
```
Comment (bastiandoetsch) [inline CHANGELOG.md:13]: you'll need to rebase against master :)

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**